app/services/bulk_upload/lettings/year2024/row_parser.rb

```diff
@@ -0,0 +1,1468 @@
+  attribute :field_100, :integer
+  attribute :field_101, :integer
+  attribute :field_102, :integer
+  attribute :field_103, :string
+  attribute :field_104, :string
+  attribute :field_105, :string
+  attribute :field_106, :integer
+  attribute :field_107, :integer
+  attribute :field_108, :integer
+  attribute :field_109, :integer
+  attribute :field_110, :integer
+  attribute :field_111, :integer
+  attribute :field_112, :integer
+  attribute :field_113, :integer
+  attribute :field_114, :integer
+  attribute :field_115, :integer
+  attribute :field_116, :integer
+  attribute :field_117, :integer
+  attribute :field_119, :integer
+  attribute :field_118, :decimal
+  attribute :field_120, :integer
+  attribute :field_121, :integer
+  attribute :field_122, :integer
+  attribute :field_123, :integer
+  attribute :field_124, :decimal
+  attribute :field_125, :decimal
+  attribute :field_126, :decimal
+  attribute :field_127, :decimal
+  attribute :field_128, :decimal
+  attribute :field_129, :integer
+  attribute :field_130, :decimal
+
+  validate :validate_valid_radio_option, on: :before_log
+
+  validates :field_11,
+            presence: {
+              message: I18n.t("validations.not_answered", question: "rent type"),
+              category: :setup,
+            },
+            inclusion: {
+              in: (0..5).to_a,
+              message: I18n.t("validations.invalid_option", question: "rent type"),
+              unless: -> { field_11.blank? },
+              category: :setup,
+            },
+            on: :after_log
+
+  validates :field_7,
+            presence: {
+              message: I18n.t("validations.not_answered", question: "property renewal"),
+              category: :setup,
+            },
+            on: :after_log
+
+  validates :field_8,
+            presence: {
+              message: I18n.t("validations.not_answered", question: "tenancy start date (day)"),
+              category: :setup,
+            },
+            on: :after_log
+
+  validates :field_9,
+            presence: {
+              message: I18n.t("validations.not_answered", question: "tenancy start date (month)"),
+              category: :setup,
+            },
+            on: :after_log
+
+  validates :field_10,
+            presence: {
+              message: I18n.t("validations.not_answered", question: "tenancy start date (year)"),
+              category: :setup,
+            },
+            format: {
+              with: /\A\d{2}\z/,
+              message: I18n.t("validations.setup.startdate.year_not_two_digits"),
+              category: :setup,
+              unless: -> { field_10.blank? },
+            },
+            on: :after_log
+
+  validates :field_5,
+            presence: {
+              if: proc { supported_housing? },
+              message: I18n.t("validations.not_answered", question: "scheme code"),
+              category: :setup,
+            },
+            on: :after_log
+
+  validates :field_6,
```

Comment:
why are we making validation changes here? I thought all that would be left to separate tickets...

Reply:
I think if you're referring to the location fields, it's cause we now only have use the new scheme and location fields, so we don't need the validation which determines which scheme/location/management group fields get answered, but we still want to validate the location. So this is just the simplified version of that. If you have other validations in mind that shouldn't be a part of this let me know!